packages/server/src/core/index.ts

```diff
@@ -19,3 +19,5 @@ export { createInputMiddleware, createOutputMiddleware } from './middleware';
 
 export { initTRPC } from './initTRPC';
 export * from './types';
+
+export type { RootConfig } from './internals/config';
```

Comment:
Could we rename this to `TRPCRootConfig` throughout?

Reply:
could do

but given this type is in a `@trpc/server` package, does this not implicitly scope it as a tRPC thing? i.e. I believe this is _technically_ a `TRPCServerRootConfig`, though that prefix feels redundant.

in the same vain, higher up, this file also exports
```ts
export type {
  Procedure,
  AnyProcedure,
  AnyQueryProcedure,
  // etc.
} from './procedure';
```
rather than 
```ts
export type {
  TRPCProcedure,
  TRPCAnyProcedure,
  TRPCAnyQueryProcedure,
  // etc.
} from './procedure';
```

if the idea and direction is to give these internal but exposed & inferred types that tRPC-context for later, we could also just
```ts
export type { RootConfig as TRPCRootConfig } from './internals/config';
```

happy to leave as-is or apply either changes